src/colorizer/ColorizeCanvas3D.ts

```diff
@@ -131,6 +168,41 @@ export class ColorizeCanvas3D implements IRenderCanvas {
     }
   }
 
+  private updateLineMaterial(): void {
+    if (!this.params) {
+      return;
+    }
+    const { trackPathColorMode, outlineColor, trackPathColor, trackPathWidthPx } = this.params;
+    const modeToColor = {
+      [TrackPathColorMode.USE_FEATURE_COLOR]: new Color("#ffffff"),
+      [TrackPathColorMode.USE_OUTLINE_COLOR]: outlineColor,
+      [TrackPathColorMode.USE_CUSTOM_COLOR]: trackPathColor,
+    };
+    const color = modeToColor[trackPathColorMode];
+    for (const lineObject of [this.lineObject, this.lineOverlayObject]) {
+      lineObject.setColor(color, trackPathColorMode === TrackPathColorMode.USE_FEATURE_COLOR);
+      lineObject.setLineWidth(trackPathWidthPx);
+    }
+  }
+
+  private updateLineGeometry(points: Float32Array, colors: Float32Array): void {
+    if (!this.params || !this.params.track || !this.params.dataset) {
+      return;
+    }
+    if (!this.view3d.hasLineObject(this.lineObject)) {
+      this.view3d.addLineObject(this.lineObject);
+      this.view3d.addLineObject(this.lineOverlayObject);
```

Comment:
Maybe odd to be relying on three's `Line3d` type for the public API of vole-core. Fine with it for now, since this is for the moment an odd extra corner of the API; maybe fine with it forever, honestly. But worth thinking about for the future.

Reply:
oh duh 🤦